src/incremental/compareAST.ml

```diff
@@ -5,6 +5,41 @@ type global_type = Fun | Decl | Var
 
 and global_identifier = {name: string ; global_t: global_type} [@@deriving ord]
 
+type method_rename_assumption = {original_method_name: string; new_method_name: string; parameter_renames: (string, string) Hashtbl.t}
+type method_rename_assumptions = (string, method_rename_assumption) Hashtbl.t
+
+(*rename_mapping is carried through the stack when comparing the AST. Holds a list of rename assumptions.*)
```

Comment:
Any global state will make it impossible to parallelize parsing/merging/comparing, so not introducing new global state into these parts would be good, if we hope some day to be able to speed up our preprocessing.

There is another `(typ * typ) list` structure being passed around in many of the comparison functions though, so with this change there would be two. Since most recursion here just passes them around, they could be packaged together into a single record type, which is also easier to extend in the future with other structures.

Reply:
The carrying through the stack is required for the detection of renamed functions and global variables which I am also working on. As such I am currently working on making the whole AST comparison fully functional without side effects.